### PR TITLE
[feat] : make `txid` and `best_block` of `Hash` type

### PR DESCRIPTION
**Pull request by TAdev0**
- [x] issue #71 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests

almost done, just need to regenerate blocks to make it compile

### CODE REVIEW

File: src/validation.cairo [hunk 1]
```diff
@@ -229,7 +229,7 @@ fn validate_coinbase(block: @Block, total_fees: u64, block_height: u32) -> Resul
     assert(*tx.inputs[0].previous_output.vout == 0xFFFFFFFF, 'vout should be 0xFFFFFFFF');
 
     // Ensure the input's TXID is zero
-    assert(*tx.inputs[0].previous_output.txid == 0, 'txid should be 0');
+    assert(*tx.inputs[0].previous_output.txid == Hash { value: [0; 8] }, 'txid should be 0');
```
Comment (TAdev0) [inline src/validation.cairo:232]: here is actually a case where we could have this `new` function that generates a default Hash.

@maciejka  what do you prefer? leaving it this is way or adding a new method for that?

File: src/state.cairo [hunk 2]
```diff
@@ -12,62 +14,27 @@ pub struct ChainState {
...
-/// Inclusion proof for multiple leaves
-#[derive(Drop, Copy)]
-pub struct UtreexoBatchProof {
-    /// Indices of tree leaves, one for each output in the utreexo set
-    pub targets: Span<u64>,
-    /// All the nodes required to calculate the root
-    pub proof: Span<felt252>,
-}
-
 /// Represents a block in the blockchain.
```
Comment (TAdev0) [inline src/state.cairo:28]: here i'm just modifying order of structs in state.cairo to make it more logical

State -> block -> header -> Transaction -> TxIn -> OutPoint -> TxOut

File: src/utils.cairo [hunk 3]
```diff
@@ -32,6 +32,32 @@ impl HashPartialEq of PartialEq<Hash> {
     }
 }
 
+impl HashPartialOrd of PartialOrd<Hash> {
```
Comment (maciejka) [inline src/utils.cairo:35]: We want `PartialEq` not `PartialOrd`, right? What partial order one can have with hashes?

File: src/validation.cairo (hunk 1 above)
Comment (maciejka) [reply src/validation.cairo:232]: Implement Default?

**Review by maciejka — CHANGES_REQUESTED** [review]

File: src/validation.cairo (hunk 1 above)
Comment (TAdev0) [reply src/validation.cairo:232]: ok!

**Review by maciejka — APPROVED** [review]

**Pull request merged**